### DISCUSSION TITLE
QST: How to find geometries in a dynamic buffer quickly?

### DISCUSSION THREAD

**Discussion by forestbat** [geopandas/geopandas#2733] [Q&A]
I want to get the nearest `LineString` with an appointed `Point`, so I wrote these codes:
```python
# use dynamic buffer to find linestrings
xmin, ymin, xmax, ymax = (point.buffer(initial_buffer)).bounds         
gpd_df_clip = gpd_line_series.clip_by_rect(xmin, ymin, xmax, ymax)     
while gpd_df_clip.empty:                 
    initial_buffer = initial_buffer + buffer_steps                     
    xmin, ymin, xmax, ymax = (point.buffer(initial_buffer)).bounds     
    gpd_df_clip = gpd_line_series.clip_by_rect(xmin, ymin, xmax, ymax) 
# from gpd_df_clip find the nearest line
min_dist = 360                                                         
min_dis_line = gpd_df_clip.geometry[0]                                 
for line in gpd_df_clip.geometry:                                      
    dis = point.distance(line)                                         
    if dis < min_dist:                                                 
        min_dist = dis                                                 
        min_dis_line = line                                            
```
But when I test the code I knew `clip_by_rect` will return a series or dataframe which contains `GEOMETRYCOLLECTION EMPTY` (Note: empty geometries or geometries that do not overlap with the specified bounds will result in GEOMETRYCOLLECTION EMPTY), so I probably should write:
`while gpd_df_clip.all() is GEOMETRYCOLLECTION EMPTY:`
However I don't know if this is a slow operation, so are there any other efficient method to complete my task?

**Comment by martinfleis** [comment]
Would `sjoin_nearest` solve your problem? https://geopandas.org/en/stable/docs/reference/api/geopandas.sjoin_nearest.html
